Schema.php

```diff
@@ -449,7 +449,35 @@ protected function getColumnPhpType($column)
      */
     protected function findColumns($index)
     {
-        $sql = 'DESCRIBE ' . $this->quoteSimpleIndexName($index->name);
+        $columns = $this->getIndexColumns($index->name);
+        if (!empty($columns[0]['Agent'])) {
+            $columns = $this->getIndexColumns($columns[0]['Agent']);
+        }
+
+        if (empty($columns)) {
+            return $columns;
+        }
+
+        foreach ($columns as $info) {
+            $column = $this->loadColumnSchema($info);
+            $index->columns[$column->name] = $column;
+            if ($column->isPrimaryKey) {
+                $index->primaryKey = $column->name;
+            }
+        }
+
+        return true;
+    }
+
+
+    /**
+     * Return columns for index $name
+     * @param $name string index name
+     * @return array|bool
+     * @throws \Exception
+     */
+    protected function getIndexColumns($name) {
```

Comment:
`getIndexColumn` is not a good name for this method. Especially comparing with `findColumns`.
Also code style is broken.